integration/query_projection_test.go

```diff
@@ -0,0 +1,122 @@
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/FerretDB/FerretDB/integration/shareddata"
+)
+
+func TestQueryProjection(t *testing.T) {
+	t.Parallel()
+	providers := []shareddata.Provider{shareddata.Composites}
+	ctx, collection := setup(t, providers...)
+
+	_, err := collection.InsertMany(ctx, []any{
+		bson.D{
+			{"_id", "document-composite-2"},
+			{"value", bson.A{
+				bson.D{{"field", int32(42)}},
+				bson.D{{"field", int32(44)}},
+			}},
+		},
+	})
+	require.NoError(t, err)
+
+	for name, tc := range map[string]struct {
+		projection any
+		filter     any
+		expected   bson.D
+	}{
+		"FindProjectionInclusions": {
+			filter: bson.D{{"_id", "document-composite"}},
+			// TODO: fix projection
+			projection: bson.D{{"foo", int32(1)}, {"42", true}},
+			expected:   bson.D{{"_id", "document-composite"}},
+		},
+		"FindProjectionExclusions": {
+			filter: bson.D{{"_id", "document-composite"}},
+			// TODO: fix projection
+			projection: bson.D{{"foo", int32(0)}, {"array", false}},
+			expected:   bson.D{{"_id", "document-composite"}, {"value", bson.D{{"foo", int32(42)}, {"42", "foo"}, {"array", bson.A{int32(42), "foo", nil}}}}},
+		},
+		"FindProjectionIDExclusion": {
+			filter: bson.D{{"_id", "document-composite"}},
+			// TODO: fix projection
+			projection: bson.D{{"_id", false}, {"array", int32(1)}},
+			expected:   bson.D{},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			cursor, err := collection.Find(
+				ctx,
+				tc.filter,
+				options.Find().SetProjection(tc.projection),
```

Comment:
That seems to be short enough to put on one line. One of the `expectred` documents above is longer

Reply:
Fixed